helper/parseItems.js

```diff
@@ -1,120 +1,143 @@
-    const processCakeBags = async (items) => {
-        for (const categoryItems of Object.values(items)) {
-            for (const item of categoryItems) {
-                if (!item?.tag?.ExtraAttributes?.new_year_cake_bag_data) {
-                    continue;
-                }
-
+    await Promise.all([
+        // Process new year cake bags
+        (async () => {
+            // Get the new years cake bags
+            const allItems = Object.values(items)
+                .filter(Array.isArray)
+                .flat()
+                .filter((item) => item?.tag?.ExtraAttributes?.new_year_cake_bag_data);
+            // Decode the cakes from the new years cake bags data
+            for (const item of allItems) {
                 const cakeBagData = await decodeItem(Buffer.from(item.tag.ExtraAttributes.new_year_cake_bag_data, 'base64'));
                 item.tag.ExtraAttributes.new_year_cake_bag_years = cakeBagData
                     .filter((cake) => cake.id && cake.tag?.ExtraAttributes?.new_years_cake)
                     .map((cake) => cake.tag.ExtraAttributes.new_years_cake);
             }
-        }
-    };
-
-    await Promise.all([
-        processCakeBags(items),
+        })(),
+        // Add the sacks inventory
         (() => {
+            // Get the data
             const sacksData = profileData.sacks_counts || profileData.inventory?.sacks_counts;
-            items.sacks = sacksData
-                ? Object.entries(sacksData)
-                      .filter(([_, amount]) => amount)
-                      .map(([id, amount]) => ({ id, amount }))
-                : [];
+            // Process the data
+            if (sacksData) {
+                items.sacks = Object.entries(sacksData)
+                    .filter(([_, amount]) => amount)
+                    .map(([id, amount]) => ({ id, amount }));
+            } else {
+                items.sacks = [];
+            }
         })(),
+        // Add the essence inventory
         (() => {
-            items.essence = profileData.currencies?.essence
-                ? Object.entries(profileData.currencies.essence).map(([id, data]) => {
-                      return {
-                          id: `ESSENCE_${id.toUpperCase()}`,
-                          amount: data.current,
-                      };
-                  })
-                : [];
+            // Get the data
+            const essenceData = profileData.currencies?.essence;
+            // Process the data
+            if (essenceData) {
+                items.essence = Object.entries(essenceData).map(([id, data]) => ({ id: `ESSENCE_${id.toUpperCase()}`, amount: data.current }));
+            } else {
+                items.essence = [];
+            }
         })(),
+        // Add the pets inventory
         (() => {
+            // Process the data
             items.pets = profileData.pets_data?.pets?.map((pet) => ({ ...pet })) ?? [];
         })(),
```

Comment:
Is there really any benefit to having all of these be in the `Promise.all` since there is no `await`s?